### ISSUE TITLE
SVG

### ISSUE THREAD

**Issue by ways2read** [ways2read/InDesignA11Y#9]
Knowledge base reference: http://kb.daisy.org/publishing/docs/html/svg.html

**Comment by LauraB7** [comment]
User story: "As a low vision user I am able to enlarge a map (for example) to a high level without degradation of the image because it is included as an SVG."

**Comment by raman211** [comment]
@gregoriopellegrino @jonaslil @ways2read @LauraB7 @gilbertconsult
This is Ramanjulu from Adobe.
Could you please provide detailed information on the required things for the accessibility of SVG? 
CC: @NawneetG @Racmathu

**Comment by jonaslil** [comment]
## Current situation

### SVGs are rasterized by default

InDesign supports placing SVG images, but the default EPUB export behaviour is to convert them to PNG. This rasterizes the image, negating the accessibility benefits of SVG.

To preserve SVGs in EPUB export, users must unselect "Preserve appearance from layout (for graphic/media objects)" in the export dialogue, which applies globally to all images in the EPUB. Alternatively, users can adjust this for individual images via Object export options > EPUB and HTML, changing “Preserve appearance from layout” to “Use existing image for graphic objects”. However, these settings can have unwanted consequenses, discouraging real-life use of SVG in EPUBs.

When SVGs are preserved, InDesign offers two export methods: Embed Code and Object Tags.

### Adding image descriptions isn't possible

Alt text added to an SVG or marking it as a decorative image in Object export options currently has no effect on EPUB output.

## Proposed solution

### Support `<img>` export for SVGs

InDesign should export SVGs using the HTML `<img>` tag by default, without rasterizing them. This allows for adding image descriptions via the alt attribute and using `role="presentation"` for decorative images.

- **Default behavior:** When SVGs are placed, `<img>` export should be the default behaviour. SVGs should not be rasterized.
- **Image descriptions:** Alt text for SVGs should be exported using the `alt` attribute, as with other images.
- **Decorative SVGs:** For SVGs marked as decorative, `alt=""` and `role="presentation"` should be applied.

### Retain existing options

We do not propose substantive changes to the current Embed Code and Object Tags options. However, we have noticed some issues with these methods that cause EPUB validation errors. We will report these separately.

**Comment by raman211** [comment]
Hi @jonaslil 
In case of SVG in Graphic frame of InDesign document, has appearance effects applied on it like stroke/fill/opacity etc, but from the Object Export Options the default behaviour is chosen
Do you still suggest SVG being replaced as an SVG?
The current behaviour is that we convert it to PNG and apply effects to it.
CC: @ways2read @LauraB7 @gregoriopellegrino

**Comment by jonaslil** [comment]
Yes, the default behavior should still be to export the image as an SVG, even if that means effects (such as stroke, fill, or opacity) applied in InDesign are not preserved.

The option to convert the image to PNG and retain those effects would still be available for users who need it.

At the moment, if you uncheck "Preserve Appearance from Layout (for Graphic/Media objects)" and choose either "Embed Code" or "Object Tag" for the SVG, those effects are already lost - it's just not the default behavior.

**Comment by jonaslil** [comment]
If making export as an SVG the default behaviour is not an option, as indicated in the latest call, we propose the following:
 
Retain the current default behaviour, where SVG images are exported as PNG. This option already supports alt texts or marking images as decorative in the usual way. It also preserves visual effects applied in InDesign.
 
Add a new export option for SVG, named "img tag" (or similar), in addition to the currently available choices: "Embed code" and "Object Tags". This new option would be added to:
 
- the Object Export Options window, EPUB and HTML tab, "Use SVG as" dropdown
- the EPUB Reflowable layout export options window, Object tab, "Use SVG as" dropdown
 
When the user selects "img tag", the SVG file is retained in its original format and not converted to PNG. It remains as a separate file, referenced using a standard HTML `<img>` tag.
 
When the `<img>` tag is used:
 
- Alt text provided by the user should be exported as it is for other image formats.
- If the user selects "Decorative image (no alt text)", the attribute `role="presentation"` should be added to the `<img>` element.
 
Code examples
 
SVG with alt text:

```html
<img alt="Some alternative text" src="image/my_graphic.svg"/>
```
 
Decorative SVG:

```html
<img alt="" role="presentation" src="image/my_graphic.svg"/>
```

**Comment by jonaslil** [comment]
@raman211, @racmathu

In our tests of the current prerelease build, the new Image Tags option works as expected when applied to individual images via the Object Export Options window.
 
However, when SVG export settings are applied globally via the EPUB Reflowable Layout Export Options window, there is an inconsistency.
 
If the user unchecks “Preserve Appearance from Layout (for Graphic/Media objects” without also selecting “Use Existing Image for Graphic Objects”, the “Use SVG as” menu becomes active, and Image Tags can be selected – just like Embed Code and Object Tags.

![Image](https://github.com/user-attachments/assets/849e6b3d-5ee2-4ff3-91fa-37e1fddd8733)

 
While Embed Code and Object Tags work in this case, Image Tags does not: it produces the same output as Embed Code. This gives the impression that Image Tags isn’t really functional.
 
For Image Tags to work, the user must also select “Use Existing Image for Graphic Objects”.

![Image](https://github.com/user-attachments/assets/1d16908f-672b-48b0-bff7-90b972f9609b)
 
**Proposed fix:**
When the user unchecks “Preserve Appearance from Layout (for Graphic/Media objects” and selects “Use SVG as: Image Tags”, the output should reflect the Image Tags setting – even if "Use Existing Image for Graphic Objects" is not selected.

We also propose making Image Tags the default option in the “Use SVG as” menu. The current default, Embed Code, is impractical for EPUB: it bloats the XHTML, doesn’t preserve alt text, and causes validation errors.

**Comment by raman211** [comment]
@jonaslil We have crossed the timelines of ID20.4
We will try to address these in ID20.5 timelines.

**Comment by Nainabhandari24** [comment]
Hi @jonaslil ,

We have fixed this issue on ID20.5.30 pre-release build. 
Could you please assist in verify the reported issue on the ID20.5.0.30 pre-release build? Your support in confirming would be greatly appreciated.

CC: @raman211 

Thanks,
Naina

**Comment by Nainabhandari24** [comment]
Hi @jonaslil,

We have successfully completed the go-live of ID20.5.0 across all regions. Kindly proceed with validating the reported issue on the ID20.5.0 build.

Thanks,
Naina

**Comment by jonaslil** [comment]
@Nainabhandari24, thanks for the update. I can confirm that the reported issue is solved in the 20.5 release. I’ll go ahead and close this issue now.
